### PR TITLE
LPS-88329 [FE] - Make Sidebar dynamic per contributor

### PR DESCRIPTION
**Pull request by thektan**
https://issues.liferay.com/browse/LPS-88329

Hey Andrés, it seems like you might missed a handful of comments from me and Bryce because of Github collapsing our comments which look like:

screenshot 2019-01-10 11 54 25

---

To save some additional back and forth with minor formatting changes, I went ahead and made some of the changes. 👍

**Changes:** 3e9edf0...5c1a5f4

Could you review my changes and lastly:
- [x] Make the change noted in this comment: https://github.com/bryceosterhaus/liferay-portal/pull/13#discussion_r246486843
- [x] Double check the comments made on https://github.com/bryceosterhaus/liferay-portal/pull/13 and https://github.com/thektan/liferay-portal/pull/14
- [x] Make sure the root conjunction button changes when clicked, which likely got broken during all these review changes 😕

### CODE REVIEW

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js [hunk 1]
```diff
@@ -67,30 +67,19 @@ class ContributorBuilder extends React.Component {
...
-								} : contributor;
-							}
-						)
-					};
-				}
-
-				return newState;
-			}
-		);
+		if (this.state.editing === index) {
```
Comment (andresfulla) [inline modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js:70]: I ran into really weird problems in the past relying on `this.state` values at `this.setState`, same with props. It is recommended to pass a function.
More info here: https://reactjs.org/docs/faq-state.html#why-is-setstate-giving-me-the-wrong-value

**Comment by andresfulla** [comment]
Many thanks @thektan ! Spot on! Totally missed those comments.

The only thing I'm not into is the use of `this.state` values inside `this.setState`.

File: modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js (hunk 1 above)
Comment (thektan) [reply modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/ContributorBuilder.es.js:70]: Nice catch! I'll keep this in mind for the future as well.

**Comment by thektan** [comment]
New pr: https://github.com/thektan/liferay-portal/pull/15

**Pull request closed without merging**